lib/spack/spack/test/container/cli.py

```diff
@@ -9,8 +9,8 @@
 containerize = spack.main.SpackCommand('containerize')
 
 
-def test_command(configuration_dir, capsys):
+def test_command(config, container_config_dir, capsys):
```

Comment:
Because if the user has some crazy things in their configuration, they will see spurious test failures. It's not to protect the user from this test, it's to protect the test from the user.

I found this because I had some very weird config settings (like removing the default template dirs) while I was testing this PR.

Reply:
My point is that these tests were ensuring that with the default configuration `spack containerize` works (default configuration is what is used in the CI on Github). Now they ensure that `spack containerize` works with the mock configuration, which is slightly weaker imo.

Reply:
The mock configuration uses the default configuration as well. But what the tests were doing was ensuring that they worked in whatever environment the user had, not the default environment.

Reply:
Maybe we should have an additional fixture `default_configs`, but it seems unnecessary since the `config` fixture is the default configs plus 3 empty directories.

Reply:
> The mock configuration uses the default configuration as well.

Doesn't it swap for what is in `test/data/config`? Anyhow, I see the two different point of views:
1. Using the default non-mock configuration and targeting a :heavy_check_mark: on our public CI (might :negative_squared_cross_mark: in a customized environment)
2. Using the mock configuration so that the test :heavy_check_mark:  even in customized environments

My point is that 2. will be :heavy_check_mark: even if we were to add something weird to our defaults, because they would be masked by the mock configuration.

Reply:
I'll add a fixture that replaces the user configs with the default configs.